engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java

```diff
@@ -326,8 +326,8 @@ public RID next() {
           // MERGE VALUES
           final RID[] newArray = Arrays.copyOf(currentValues, currentValues.length + tempCurrentValues.length);
 
-          for (int k = currentValues.length; k < newArray.length; ++k)
-            newArray[k] = tempCurrentValues[k - currentValues.length];
+            if (newArray.length - currentValues.length >= 0)
+                System.arraycopy(tempCurrentValues, currentValues.length - currentValues.length, newArray, currentValues.length, newArray.length - currentValues.length);
```

Comment:
It looks like the 2nd parameter `currentValues.length - currentValues.length` is wrong. Shouldn't be just `currentValues.length` ?

Reply:
agreed!

Reply:
Also, I think the if should be
```
            if (newArray.length - currentValues.length > 0)
```

Reply:
Ok, I can change a little bit that block with 
```java
        if (i == 0 || currentValues == null)
          currentValues = tempCurrentValues;
        else if (tempCurrentValues.length > 0) {
          // MERGE VALUES
          final RID[] newArray = Arrays.copyOf(currentValues, currentValues.length + tempCurrentValues.length);
          System.arraycopy(tempCurrentValues, currentValues.length, newArray, currentValues.length, newArray.length - currentValues.length);
          currentValues = newArray;
        }
```